android-components/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt

```diff
@@ -27,17 +27,23 @@ open class FullScreenFeature(
     private val fullScreenChanged: (Boolean) -> Unit,
 ) : LifecycleAwareFeature, UserInteractionHandler {
     private var scope: CoroutineScope? = null
+    private var isFullscreenCustomTab: Boolean = false
```

Comment:
We have a concept of state holding here in this feature one line below - the Observation. We can add extra state into that instead of creating a new concept of state holding. See below.